eng/BootStrapMsBuild.props

```diff
@@ -1,20 +1,23 @@
 <Project>
 
-  <!--
-  Construct a location of MSBuild bootstrap folder - to be used for deployment and for tests
-   relying on bootstrapped MSBuild
-   -->
+  <!-- Construct a location of MSBuild bootstrap folder - to be used for deployment and for tests relying on bootstrapped MSBuild -->
+
+  <PropertyGroup Condition="!$(TargetFramework.StartsWith('net4'))">
+    <NetVersion>8.0.302</NetVersion>
```

Comment:
If this needs to be hardcoded - then it probably should go to Version.props, plus release-checklist.md should have an entry for updating this.

Reply:
Or actually probably not a release-checklist.md, but checklist for .NET retargetting (that we do not yet have documented, but we should document it. Previous efforts: https://github.com/dotnet/msbuild/issues/8969, https://github.com/dotnet/msbuild/pull/7790)